### PR TITLE
Decorate class

### PR DESCRIPTION
**Pull request by brhanuhailu**
### In this pull request I:

- [X] Create a class Nameable.
   - Implement a method called correct_name that will raise a NotImplementedError.
- [X] Turn Person class to Nameable
   - class has a method correct_name implemented. 
- [X] Prepared base Decorator
   - Implemented the correct_name method that returns the result of the correct_name method of the @nameable
- [X] Prepare CapitalizeDecorator and TrimmerDecorator
   - For the CapitalizeDecorator:
       - Create a class that inherits from the base Decorator class.
       - Implemented a method correct_name that capitalizes the output of @nameable.correct_name
  - For the TrimmerDecorator:
      - Create a class that inherits from the base Decorator class.
      - Implement a method correct_name that makes  output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.
- [X] Tested decorate your person

### CODE REVIEW

File: decorator/trimmer_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require_relative 'base'
+
+class TrimDecorator < Decorator
+  def correct_name
+    super.slice(0, 9) if super.length > 10
+  end
+end
```
Comment (ShadyShawkat) [inline decorator/trimmer_decorator.rb]: - [ ] This method returns the right result but it returns a result **_only_** if `super.length > 10`. That means if you passed a string that is less than 10 characters long, it won't return anything. Kindly fix this behavior.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @brhanuhailu, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: decorator/trimmer_decorator.rb (hunk 1 above)
Comment (brhanuhailu) [reply decorator/trimmer_decorator.rb]: @ShadyShawkat Thank you for your constructive feedback.  just solved

**Review by roshan-bajgain — APPROVED** [review]
Hi @brhanuhailu, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️
- Good job on implementing the changes ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**